### ISSUE TITLE
Right for the right reasons: Training differentiable models by constraining their explanations

### ISSUE THREAD

**Issue by richardtomsett** [dais-ita/interpretability-papers#43]
[Right for the right reasons: Training differentiable models by constraining their explanations](https://www.ijcai.org/proceedings/2017/371)
Expressive classifiers such as neural networks are among the most accurate supervised learning methods in use today, but their opaque decision boundaries make them difficult to trust in critical applications. We propose a method to explain the predictions of any differentiable model via the gradient of the class label with respect to the input (which provides a normal to the decision boundary). Not only is this approach orders of magnitude faster at identifying input dimensions of high sensitivity than sample-based perturbation methods (e.g. LIME), but it also lends itself to efficiently discovering multiple qualitatively different decision boundaries as well as decision boundaries that are consistent with expert annotation. On multiple datasets, we show our approach generalizes much better when test conditions differ from those in training.

Bibtex:
@inproceedings{ijcai2017-371,
  author    = {Andrew Slavin Ross, Michael C. Hughes, Finale Doshi-Velez},
  title     = {Right for the Right Reasons: Training Differentiable Models by Constraining their Explanations},
  booktitle = {Proceedings of the Twenty-Sixth International Joint Conference on
               Artificial Intelligence, {IJCAI-17}},
  pages     = {2662--2670},
  year      = {2017},
  doi       = {10.24963/ijcai.2017/371}
}

**Comment by richardtomsett** [comment]
Another recent method, due to Ross et al. (2017) and inspired by LIME*, allows a developer to constrain their model during training to be “right for the right reasons” (RRR) rather than learning spurious correlations in the training data. RRR works using binary masks that specify whether an input feature should be irrelevant to the classification of that example, as assessed by a human expert. The authors additionally propose an automatic method that learns a set of models using different masks designed to create models with different decision boundaries. This automated approach provides a way of quantifying ambiguity in the training data by counting the number of different models that can be learned without reducing accuracy.

*see issue #41
